db/migrate/20220208134539_rename_cent_os_stream_os.rb

```diff
@@ -0,0 +1,13 @@
+class RenameCentOsStreamOs < ActiveRecord::Migration[6.0]
+  def up
+    User.without_auditing do
+      Operatingsystem.unscoped.where("type = 'Redhat' and name = 'CentOS' and (description like '%CentOS Stream 8%' or title like '%CentOS Stream 8%')").update_all(name: "CentOS_Stream", title: "CentOS_Stream 8", description: "CentOS_Stream 8")
```

Comment:
Do we need to update the description and title?
The issue is fixed by only changing the name.

Reply:
Well name is not the culprit actually, see the original report exception: `Backtrace for 'Action failed' error (ActiveRecord::RecordInvalid): Validation failed: Description has already been taken, Title has already been taken`

Foreman operating system model class design:
* has a name, major, minor fields with unique constraint in the scope of name-major-minor
* has a title field that is automatically generated before validation as follows: "name major(.minor)"
* title must be also unique as it is used as "friendly_id"
* there is also a description which overrides title if set by a user and it is also unique
* when any facts check-in, they try to update OS name, major, minor causing title and description to be regenerated
* I can see that description is set automatically on fact upload too but I cannot find the place where it actually gets set

My conclusion: 42
Don't panic, keep a towel nearby.

Reply:
Correction, I can see where description is set, it is actually explicitly in the fact parser. Title is auto-generated by OS. Well, "makes sense".

Reply:
> The issue is fixed by only changing the name.

Or maybe I am not following what you mean?